domain/model/errors/errors.go

```diff
@@ -34,6 +34,10 @@ const (
 	// a model is not considered valid.
 	CredentialNotValid = errors.ConstError("model credential not valid")
 
+	// CredentialNotFound describes an error that occurs when the credential for
+	// a model is not found.
+	CredentialNotFound = errors.ConstError("credential not found")
```

Comment:
I know you won't enjoy this but can you delete this as it is a double up and use the NotFound error from the credential domain.